### PR TITLE
W-10875252: Implement the Unit of Works Extension for the Profiling API

### PR DESCRIPTION
**Pull request by IvanAndresFritzler**

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/internal/profiling/consumer/tracing/span/DefaultSpanDuration.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import org.mule.runtime.api.profiling.tracing.SpanDuration;
+
+/**
+ * Dafault implementation for {@link SpanDuration}
+ */
+public class DefaultSpanDuration implements SpanDuration {
+
+  private final Long startTime;
+  private final Long endTime;
+
```
Comment (marianogonzalez) [inline core/src/main/java/org/mule/runtime/core/internal/profiling/consumer/tracing/span/DefaultSpanDuration.java:20]: shouldn't this be native?

File: core/src/main/java/org/mule/runtime/core/internal/profiling/consumer/tracing/span/ExecutionSpan.java [hunk 2]
```diff
@@ -20,21 +22,19 @@
...
-    this.endTime = endTime;
+    this.spanDuration = spanDuration;
     this.parent = parent;
   }
 
   @Override
-  public Span getParent() {
-    return parent;
+  public Optional<Span> getParent() {
+    return Optional.ofNullable(parent);
```
Comment (marianogonzalez) [inline core/src/main/java/org/mule/runtime/core/internal/profiling/consumer/tracing/span/ExecutionSpan.java:37]: static import

**Review by marianogonzalez — CHANGES_REQUESTED** [review]

**Review by fsgonz — APPROVED** [review]

**Review by marianogonzalez — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/mule/runtime/core/internal/profiling/consumer/tracing/span/DefaultSpanDuration.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ * Dafault implementation for {@link SpanDuration}
+ */
+public class DefaultSpanDuration implements SpanDuration {
+
+  private final long startTime;
+  private final long endTime;
+
+  public DefaultSpanDuration(long startTime, long endTime) {
+    this.startTime = startTime;
+    this.endTime = endTime;
```
[the remark this answers is not in the document]
Comment (IvanAndresFritzler) [reply core/src/main/java/org/mule/runtime/core/internal/profiling/consumer/tracing/span/DefaultSpanDuration.java:21]: I think that only receiving the start timestamp is better given how things ended up being implemented. At the beginning. I will modify it to something like your suggestion.

[the remark this answers is not in the document]
Comment (marianogonzalez) [reply core/src/main/java/org/mule/runtime/core/internal/profiling/consumer/tracing/span/DefaultSpanDuration.java:21]: Cool. Note however that my comment intends to go beyond the constructor but the overall design choices of what should be immutable and what shouldn't. Regardless of the constructor and inner representation, when you do end the span the duration instance gets overriden. The ask is to re-evaluate but most importantly document that design choice.

[the remark this answers is not in the document]
Comment (IvanAndresFritzler) [reply core/src/main/java/org/mule/runtime/core/internal/profiling/consumer/tracing/span/DefaultSpanDuration.java:21]: I will add more detail about this to the SPEC.

**Pull request closed without merging**